src/backend/metal/src/command.rs

```diff
@@ -1050,11 +1046,14 @@ impl CommandSink {
     }
 }
 
+const WIDE_INDEX_BIT: buffer::Offset = 1u64<<63;
+
 #[derive(Clone, Copy, Debug)]
 pub struct IndexBuffer<B> {
     buffer: B,
-    offset: buffer::Offset,
-    index_type: MTLIndexType,
+    // Combining the index type in here allows us to reduce the `soft::RenderCommand` size
+    offset_and_index_type: buffer::Offset,
```

Comment:
It would be great if we could still use the `MTLIndexType` directly, but it's 64 bits... So by splitting this into 2 fields we'd avoid having the bit magic, but the cast is still going to be there. I don't have a strong preference either way

Reply:
Yeah I think the reducing the bit magic is good